app/models/user.rb

```diff
@@ -1,9 +1,12 @@
 class User < ApplicationRecord
-  has_secure_password 
+  has_secure_password
 
   has_many :user_recipes
   has_many :recipes
 
   validates :password_digest, presence: true
   validates :username, uniqueness: true, presence: true
+  # validates_format_of :zip,
+  #                     :with => %r{\d{5}(-\d{4})?},
+  #                     :message => "should be formatted as 12345 or 12345-1234"
```

Comment:
Same as above.

Reply:
I like that you're thinking of zips needing to be valid, but I think this would be good for an extension. (which is prolly why you have it commented out)

Reply:
Through some research on input validation for GraphQL queries and mutations, I was able to find some info on a constraint directive for GraphQL that has an integrated way to design custom validations. With that said, I agree that adding in these input validations would more of an extension.